dmutils/email/dm_mailchimp.py

```diff
@@ -120,7 +122,16 @@ def subscribe_new_emails_to_list(self, list_id, email_addresses):
                 success = False
         return success
 
-    @timeout_retry
     def get_email_addresses_from_list(self, list_id):
-        member_data = self._client.lists.members.all(list_id, get_all=True)
-        return [member["email_address"] for member in member_data["members"]]
+        email_addresses = []
+        offset = 0
+        while True:
+            member_data = self.timeout_retry(
+                self._client.lists.members.all
+            )(list_id, count=PAGINATION_SIZE, offset=offset)
+            if not member_data["members"]:
+                break
+            offset += PAGINATION_SIZE
+            email_addresses += [member["email_address"] for member in member_data["members"]]
```

Comment:
Minor quibble - `email_addresses.extend(member["email_address"] for member in member_data["members"])` will be a tad faster as it never has to build an intermediate list.

Reply:
neat!